src/h5web/dataset-visualizer/utils.ts

```diff
@@ -46,3 +47,20 @@ export function getSupportedVis(dataset?: HDF5Dataset): Vis[] {
   // Remove Raw vis if any other vis is supported
   return supported.length > 1 ? supported.slice(1) : supported;
 }
+
+export function useActiveVis(
+  supportedVis: Vis[]
+): [Vis | undefined, (vis: Vis) => void] {
+  const [activeVis, setActiveVis] = useState<Vis>();
+
+  // When switching between two datasets, `activeVis` may become stale for one render
+  const isStale = !activeVis || !supportedVis.includes(activeVis);
```

Comment:
To check whether `activeVis` is stale, we could compare the current and previous `dataset` passed as prop to `DatasetVisualizer`. This would match the beahaviour before the regression: `activeVis` would reset back to the last supported visualization when selecting a dataset.

However, a better behaviour is to not change the `activeVis` if the vusalization is also supported by the newly selected dataset. This is why I only mark `activeVis` as stale if it doesn't exist in the new `supportedVis` array.

Reply:
Just a thought: I find more readable to reverse the logic by checking if the visualization is valid rather than stale.  This line would become 
```suggestion
  const isValid = activeVis && supportedVis.includes(activeVis);
``` 
and the following checks below would be reversed or check `!isValid`.

Reply:
You're totally right 👍